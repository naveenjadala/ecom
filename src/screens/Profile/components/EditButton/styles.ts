import {StyleSheet} from 'react-native';
import {colors} from '../../../../assets/colors';

export const editButtonStyles = StyleSheet.create({
  container: {
    width: '60%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: colors.grays.gray200,
    padding: 6,
  },
  text: {
    // textAlign: 'center',
  },
});
