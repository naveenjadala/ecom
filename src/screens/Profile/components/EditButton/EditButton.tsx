import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import React from 'react';
import {editButtonStyles} from './styles';

type Props = {onClick: () => void};
/**
 * Renders an EditButton component.
 *
 * @param {Props} props - the props for the EditButton component
 * @return {JSX.Element} the rendered EditButton component
 */
const EditButton = (props: Props) => {
  return (
    <TouchableOpacity
      style={editButtonStyles.container}
      accessibilityRole="button"
      onPress={() => props.onClick()}>
      <Text style={editButtonStyles.text}>Edit Profile</Text>
    </TouchableOpacity>
  );
};

export default EditButton;

const styles = StyleSheet.create({});
