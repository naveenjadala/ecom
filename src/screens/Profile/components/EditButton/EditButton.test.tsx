import {fireEvent, render, screen, waitFor} from '../../../../test/test-utils';
import EditButton from './EditButton';

const buttonClick = jest.fn();

test('should render EditButton snapShort', () => {
  render(<EditButton onClick={buttonClick} />);
  expect(screen).toMatchSnapshot();
});

describe('should render EditButton', () => {
  test('should render touchable Image', () => {
    render(<EditButton onClick={buttonClick} />);
    const getClickableImageView = screen.getByRole('button');
    expect(getClickableImageView).toBeTruthy();
  });

  test('should fire click event', async () => {
    const buttonClick = jest.fn();
    render(<EditButton onClick={buttonClick} />);
    const getClickableImageView = screen.getByRole('button');
    fireEvent.press(getClickableImageView);
    await waitFor(() => expect(buttonClick).toHaveBeenCalledTimes(1));
  });
});
