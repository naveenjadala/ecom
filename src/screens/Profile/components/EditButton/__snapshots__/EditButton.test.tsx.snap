// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditButton snapShort 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTSafeAreaView
      style={
        {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "minHeight": "100%",
          "minWidth": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#E4E4E4",
            "borderRadius": 5,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 6,
            "width": "60%",
          }
        }
      >
        <Text
          style={{}}
        >
          Edit Profile
        </Text>
      </View>
    </RCTSafeAreaView>
  </View>
</RNCSafeAreaProvider>
`;
